driver/web/auth.go

```diff
@@ -15,450 +15,195 @@
-	logger    *logs.Logger
-}
+	tpsHandlers := tokenauth.NewHandlers(tpsAuth)
 
-func (auth *ServicesAuth) start() {
-	auth.logger.Info("ServicesAuth -> start")
-}
-
-func (auth *ServicesAuth) check(req *http.Request) (int, *tokenauth.Claims, error) {
-	claims, err := auth.tokenAuth.CheckRequestTokens(req)
+	systemAuth, err := newSystemAuth(serviceRegManager)
 	if err != nil {
-		return http.StatusUnauthorized, nil, errors.WrapErrorAction(typeCheckServicesAuthRequestToken, logutils.TypeToken, nil, err)
+		return nil, errors.WrapErrorAction(logutils.ActionCreate, "system auth", nil, err)
 	}
+	systemHandlers := tokenauth.NewHandlers(systemAuth)
 
-	if claims.Admin {
-		return http.StatusUnauthorized, nil, errors.ErrorData(logutils.StatusInvalid, "admin claim", nil)
-	}
-	if claims.System {
-		return http.StatusUnauthorized, nil, errors.ErrorData(logutils.StatusInvalid, "system claim", nil)
-	}
+	auth := Auth{services: &serviceHandlers, admin: &adminHandlers, enc: &encHandlers, bbs: &bbsHandlers, tps: &tpsHandlers, system: &systemHandlers}
 
-	err = auth.tokenAuth.AuthorizeRequestScope(claims, req)
-	if err != nil {
-		return http.StatusForbidden, nil, errors.WrapErrorAction(typeCheckScope, logutils.TypeRequest, nil, err)
-	}
-
-	return http.StatusOK, claims, nil
+	return &auth, nil
 }
 
-func (auth *ServicesAuth) getTokenAuth() *tokenauth.TokenAuth {
-	return auth.tokenAuth
-}
+// ServicesAuth
 
-func newServicesAuth(coreAPIs *core.APIs, serviceRegManager *authservice.ServiceRegManager, serviceID string, logger *logs.Logger) (*ServicesAuth, error) {
+func newServicesAuth(serviceRegManager *authservice.ServiceRegManager, serviceID string) (*tokenauth.StandardHandler, error) {
 	servicesScopeAuth := authorization.NewCasbinScopeAuthorization("driver/web/authorization_services_policy.csv", serviceID)
 	servicesPermissionAuth := authorization.NewCasbinStringAuthorization("driver/web/authorization_services_policy.csv")
 
 	servicesTokenAuth, err := tokenauth.NewTokenAuth(true, serviceRegManager, servicesPermissionAuth, servicesScopeAuth)
-
 	if err != nil {
 		return nil, errors.WrapErrorAction(logutils.ActionStart, "token auth for servicesAuth", nil, err)
 	}
 
-	auth := ServicesAuth{coreAPIs: coreAPIs, tokenAuth: servicesTokenAuth, logger: logger}
-	return &auth, nil
-}
-
-// AdminAuth entity
-type AdminAuth struct {
-	coreAPIs  *core.APIs
-	tokenAuth *tokenauth.TokenAuth
-	logger    *logs.Logger
-}
-
-func (auth *AdminAuth) start() {
-	auth.logger.Info("AdminAuth -> start")
-}
-
-func (auth *AdminAuth) check(req *http.Request) (int, *tokenauth.Claims, error) {
-	claims, err := auth.tokenAuth.CheckRequestTokens(req)
-	if err != nil {
-		return http.StatusUnauthorized, nil, errors.WrapErrorAction(typeCheckAdminAuthRequestToken, logutils.TypeToken, nil, err)
-	}
+	check := func(claims *tokenauth.Claims, req *http.Request) (int, error) {
+		if claims.Admin {
+			return http.StatusUnauthorized, errors.ErrorData(logutils.StatusInvalid, "admin claim", nil)
```

Comment:
I think we probably want to return `StatusForbidden` in all of these cases to avoid triggering unnecessary token refresh attempts. It is also more accurate since `StatusUnauthorized` should be used for missing/invalid credentials while `StatusForbidden` should be used for valid credentials that do not grant access to the requested resource.